lib/tasks/one_off.rake

```diff
@@ -257,4 +257,43 @@ namespace :one_off do
     puts "Total packages checked: #{input_tsv_file.count}"
     puts "Versions corrected: #{corrected_count}"
   end
+
+  desc "Retrieve Maven versions from ignored sources"
+  task :retrieve_maven_versions_from_ignored_sources, %i[output_file] => :environment do |_t, args|
+    maven_versions = Version.where(project_id: Project.where(platform: "Maven"))
+    # We want to retrieve the versions where the repository_sources are not solely ["Maven"] or ["Google"]
+    ignored_maven_versions = maven_versions.where.not("repository_sources = ?", ["Maven"].to_json).where.not("repository_sources = ?", ["Google"].to_json)
+    # We also want to retrieve the versions where repository_sources is nil, which aren't caught by the above
+    no_source_maven_versions = maven_versions.where("repository_sources IS NULL")
```

Comment:
I'm open to condensing these, although I had some errors with nil-checking as I constructed some queries with `.or`s. Since we only intend for this to be a one-off I wasn't super concerned with it and left them separate. Also allows us to log info about these different cases separately.

Reply:
I say leave them uncondensed.